packages/eui/src/components/datagrid/utils/col_widths.ts

```diff
@@ -79,11 +80,36 @@ export const useColumnWidths = ({
   setColumnWidth: (columnId: string, width: number) => void;
   getColumnWidth: (index: number) => number;
 } => {
+  const instance = useRef(`dataGridColumnWidths_${COLUMN_WIDTHS_CACHE.size}`);
```

Comment:
Hot dang, just fail me for a job interview because I got really confused by this line but is REALLY clever 🙈 I was thinking a randomly generated ID for each datagrid on the page would be safer, not realizing the map size would increase by one with every instance.

Would you be okay with leaving a code comment for this to clarify the cleverness? e.g.

```suggestion
  // Creates a nested map for each EuiDataGrid on the page. `.size` should increase with every instantiation
  const instance = useRef(`dataGridColumnWidths_${COLUMN_WIDTHS_CACHE.size}`);
```